src/track-scroll-depth.ts

```diff
@@ -0,0 +1,44 @@
+import { log } from '@guardian/libs';
+import { EventTimer } from './EventTimer';
+
+/**
+ * Collect commercial metrics on scroll depth
+ * Insert hidden elements at intervals of 1 viewport height
+ * then use an intersection observer to mark the time when the viewport intersects with these elements.
+ */
+const initTrackScrollDepth = () => {
+	const pageHeight = document.body.offsetHeight;
+	const intViewportHeight = window.innerHeight;
+	// how many viewports tall is the page?
+	const pageHeightVH = Math.floor(pageHeight / intViewportHeight);
+	const eventTimer = EventTimer.get();
+	eventTimer.setProperty('pageHeightVH', pageHeightVH);
+
+	const observer = new IntersectionObserver(
+		/* istanbul ignore next */
```

Comment:
Does this line exclude the following function from code coverage? And this is included because the intersection observer is mocked in the test?

If so, I wonder if it's better to leave this off? I think it's fine that this isn't tested, and aiming for 100% coverage is unrealistic, so I'd personally be happy to have a value more reflective of the true coverage of our tests. 

If it's for some other reason, please ignore me!

Reply:
Yes, `istanbul ignore next` excludes the following function from coverage since there isn't a way to test the intersection observer with jest.
The pattern I've seen in our code bases is to ignore untestable / difficult-to-test code and third party code so that our coverage numbers reflect the ratio of tested code to testable code. To me the `ignore` signals that we're happy with this function being untested whereas removing it and leaving the coverage of this file at ~75% would signify that we'd like to improve that number at some point.
I feel like the best thing to do would be to test this functionality with cypress tests where we wouldn't need to mock intersection observer at all, but I think setting up this repo's first cypress tests for this feature might be overkill.

Reply:
I see, I hadn't thought of the distinction between testable code and the source as a whole. Happy to leave this in then!